src/ast/data_type.rs

```diff
@@ -219,6 +219,8 @@ pub enum DataType {
     /// [hive]: https://docs.cloudera.com/cdw-runtime/cloud/impala-sql-reference/topics/impala-struct.html
     /// [bigquery]: https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#struct_type
     Struct(Vec<StructField>),
+    /// Specific to SQLite: no type specified (no coercion)
```

Comment:
Can you also please add some of the context (e.g. the example) from the PR's description to this comment? 

Perhaps you can also note it is not supported by Generic dialect as well